presto-cassandra/src/main/java/com/facebook/presto/cassandra/Murmur3PartitionerTokenRing.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cassandra;
+
+import java.math.BigInteger;
+
+import static java.math.BigInteger.ONE;
+
+public final class Murmur3PartitionerTokenRing
+        implements TokenRing
+{
+    public static final Murmur3PartitionerTokenRing INSTANCE = new Murmur3PartitionerTokenRing();
+
+    private static final BigInteger MIN_TOKEN = BigInteger.valueOf(Long.MIN_VALUE).subtract(ONE);
+    private static final BigInteger MAX_TOKEN = BigInteger.valueOf(Long.MAX_VALUE);
+    private static final BigInteger TOTAL_TOKEN_COUNT = MAX_TOKEN.subtract(MIN_TOKEN);
+
+    private Murmur3PartitionerTokenRing() {}
+
+    @Override
+    public double getRingFraction(String start, String end)
+    {
+        return getTokenCountInRange(start, end).doubleValue() / TOTAL_TOKEN_COUNT.doubleValue();
+    }
+
+    @Override
+    public BigInteger getTokenCountInRange(String startToken, String endToken)
+    {
+        long start = Long.parseLong(startToken);
```

Comment:
Do you need to special case the fact that if startToken=endToken and it's the minimum token, it's the whole ring? https://docs.datastax.com/en/drivers/java/2.0/index.html?com/datastax/driver/core/TokenRange.html

Probably this also applies above where you're checking for empty TokenRanges.